### PR TITLE
Multiple doc notes

### LINKED ISSUE #1202: on macos reregister and deregister return ok if the event source has not  been previously registered
According to reregister docs:
```
    /// The event source must have previously been registered with this instance
    /// of `Poll` otherwise the call to `reregister` will return with an error.
```

I noticed this issue when testing #1201. Macos does not follow doc constraints for reregister and deregister. The following tests fail only on macos:

running 14 tests
test is_send_and_sync ... ok
test drop_cancels_interest_and_shuts_down ... ok
test poll_erroneous_registration ... ok
test poll_registration ... ok
test register_deregister_multiple_poll_instances ... ok
test register_reregister_token_usage ... ok
test registry_behind_arc ... ok
test poll_closes_fd ... ok
test reregister_without_register ... FAILED
test add_then_drop ... ok
test reregister_overrides_interest ... ok
test zero_duration_polls_events ... ok
test deregister_without_register ... FAILED
test run_once_with_nothing ... ok

failures:

---- reregister_without_register stdout ----
thread 'reregister_without_register' panicked at 'called `Result::unwrap_err()` on an `Ok` value: ()', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

---- deregister_without_register stdout ----
thread 'deregister_without_register' panicked at 'called `Result::unwrap_err()` on an `Ok` value: ()', src/libcore/result.rs:1165:5

failures:
    deregister_without_register
    reregister_without_register

test result: FAILED. 12 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--test poll'

### PR DESCRIPTION
**Pull request by dtacalau**
reregister and deregister are thread-safe
undefined behavior for reregister and deregister w/o prev register
undefined behavior to register a source with another poll instance after deregister

Fixes #1202 and #1203.

### CODE REVIEW

File: src/poll.rs [hunk 1]
```diff
@@ -567,9 +570,15 @@ impl Registry {
...
     ///
+    /// The event source must have previously been registered with this instance
+    /// of `Poll`, otherwise the behavior is undefined.
+    ///
     /// A handle can be passed back to `register` after it has been
     /// deregistered; however, it must be passed back to the **same** `Poll`
-    /// instance.
+    /// instance, otherwise the behavior is undefined.
+    ///
+    /// This function is **thread safe**. It can be called concurrently from
```
Comment (Thomasdezeeuw) [inline src/poll.rs]: Same as above.

File: src/poll.rs [hunk 2]
```diff
@@ -502,12 +502,15 @@ impl Registry {
...
     ///
     /// The event source must have previously been registered with this instance
-    /// of `Poll` otherwise the call to `reregister` will return with an error.
+    /// of `Poll`, otherwise the behavior is undefined.
     ///
     /// See the [`register`] documentation for details about the function
     /// arguments and see the [`struct`] docs for a high level overview of
     /// polling.
     ///
+    /// This function is **thread safe**. It can be called concurrently from
```
[the remark this answers is not in the document]
Comment (dtacalau) [reply src/poll.rs]: removed, thanks.

File: src/poll.rs (hunk 1 above)
Comment (dtacalau) [reply src/poll.rs]: removed.

**Comment by Thomasdezeeuw** [comment]
Thanks.

**Pull request merged**